services/user-tenant-service/src/main/java/com/sourcefuse/jarc/services/usertenantservice/controller/TenantUserController.java

```diff
@@ -0,0 +1,276 @@
+      currentUser.getTenantId() == id
+    ) {
+      map =
+        tnUsrService.checkViewTenantRestrictedPermissions(
+          currentUser,
+          predicate,
+          UserView.class
+        );
+      predicate = (Predicate) map.get(CommonConstants.PREDICATE);
+    }
+    CriteriaBuilder cb = map.get(CommonConstants.BUILDER) != null
+      ? (CriteriaBuilder) map.get(CommonConstants.BUILDER)
+      : em.getCriteriaBuilder();
+    CriteriaQuery<UserView> cq = map.get(CommonConstants.CRITERIA_QUERY) != null
+      ? (CriteriaQuery) map.get(CommonConstants.CRITERIA_QUERY)
+      : cb.createQuery(UserView.class);
+    Root<UserView> root = cq.from(UserView.class);
+
+    if (predicate != null) {
+      predicate =
+        cb.and(predicate, cb.equal(root.get(CommonConstants.TENANT_ID), id));
+    } else {
+      predicate = cb.equal(root.get(CommonConstants.TENANT_ID), id);
+    }
+    cq.where(
+      predicate,
+      cb.notEqual(root.get("roleType"), CommonConstants.SUPER_ADMIN_ROLE_TYPE)
+    );
+    //doubt return UserDto
+    return new ResponseEntity<>(
+      new UserView()/*tnUsrService.getUserView(cq)*/,
+      HttpStatus.OK
+    );
+  }
+
+  @GetMapping("/view-all")
+  public ResponseEntity<Object> findAllUsers(@PathVariable("id") UUID id) {
+    List<UserView> usrList = tnUsrService.getAllUsers(id, UserView.class);
+    //nonRestrictedUserViewRepo ::doubt
+    return new ResponseEntity<>(usrList, HttpStatus.OK);
+  }
+
+  @GetMapping("/count")
+  public ResponseEntity<Object> userTenantCount(@PathVariable("id") UUID id) {
+    IAuthUserWithPermissions currentUser = getiAuthUserWithPermissions(id);
+
+    Predicate predicate = null;
+    Map<String, Object> map = new HashMap<>();
+
+    if (
+      currentUser
+        .getPermissions()
+        .contains(PermissionKey.VIEW_TENANT_USER_RESTRICTED.toString()) &&
+      currentUser.getTenantId() == id
+    ) {
+      map =
+        tnUsrService.checkViewTenantRestrictedPermissions(
+          currentUser,
+          predicate,
+          UserView.class
+        );
+      predicate = (Predicate) map.get(CommonConstants.PREDICATE);
+    }
+    CriteriaBuilder cb = map.get(CommonConstants.BUILDER) != null
+      ? (CriteriaBuilder) map.get(CommonConstants.BUILDER)
+      : em.getCriteriaBuilder();
+    CriteriaQuery<UserView> cq = map.get(CommonConstants.CRITERIA_QUERY) != null
+      ? (CriteriaQuery) map.get(CommonConstants.CRITERIA_QUERY)
+      : cb.createQuery(UserView.class);
+    Root<UserView> root = cq.from(UserView.class);
+
+    if (predicate != null) {
+      predicate =
+        cb.and(predicate, cb.equal(root.get(CommonConstants.TENANT_ID), id));
+    } else {
+      predicate = cb.equal(root.get(CommonConstants.TENANT_ID), id);
+    }
+    cq.where(
+      predicate,
+      cb.notEqual(root.get("roleType"), CommonConstants.SUPER_ADMIN_ROLE_TYPE)
+    );
+    long userCount;
+    userCount = tnUsrService.getUserView(cq).size();
```

Comment:
why cant we directly assign it

Reply:
had used this variable  in  commonConstant ,so that  it can be used in other file or for future puprpose instead of hard coding again and again .